test/uri_test.cpp

```diff
@@ -820,3 +820,67 @@ TEST(uri_test, non_opaque_path_has_double_slash) {
   EXPECT_EQ("/path/to/something/", instance.path());
   EXPECT_FALSE(instance.is_opaque());
 }
+
+TEST(uri_test, query_iterator_with_no_query) {
+  network::uri instance("http://example.com/");
+  ASSERT_FALSE(instance.has_query());
```

Comment:
I would add an additional test:
```
EXPECT_EQ(instance.query_begin(), instance.query_end());
```

Here to validate that there are no query {k, v} through the interator interface